src/trace_redaction/prune_perf_events_integrationtest.cc

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "perfetto/trace_processor/trace_processor.h"
+#include "src/base/test/status_matchers.h"
+#include "src/trace_redaction/trace_redaction_integration_fixture.h"
+#include "test/gtest_and_gmock.h"
+
+namespace perfetto::trace_redaction {
+namespace {
+constexpr auto kTrace = "test/data/trace-redaction-perf-sample.pftrace";
+
+constexpr auto kPackageName = "com.example.sampleapp";
+constexpr auto kPid = 25131;
+}  // namespace
+
+class PrunePerfEventsIntegrationTest
+    : public testing::Test,
+      protected TraceRedactionIntegrationFixure {
+ protected:
+  base::StatusOr<std::unique_ptr<trace_processor::TraceProcessor>> LoadTrace(
+      std::string trace_path) const {
+    auto raw_trace = ReadRawTrace(trace_path);
+    if (!raw_trace.ok()) {
+      return raw_trace.status();
+    }
+
+    std::unique_ptr<trace_processor::TraceProcessor> trace_processor =
+        trace_processor::TraceProcessor::CreateInstance(
+            trace_processor::Config());
+
+    auto read_buffer = std::make_unique<uint8_t[]>(raw_trace->size());
+    memcpy(read_buffer.get(), raw_trace->data(), raw_trace->size());
+
+    RETURN_IF_ERROR(
+        trace_processor->Parse(std::move(read_buffer), raw_trace->size()));
+    RETURN_IF_ERROR(trace_processor->NotifyEndOfFile());
+
+    return std::move(trace_processor);
+  }
+
+  void SetUp() override {
+    SetSourceTrace(kTrace);
+
+    TraceRedactor::Config tr_config;
+    auto trace_redactor = TraceRedactor::CreateInstance(tr_config);
+
+    Context context;
+    context.package_name = kPackageName;
+    base::Status status = Redact(*trace_redactor, &context);
+    if (!status.ok()) {
+      PERFETTO_ELOG("Redaction error: %s", status.c_message());
+    }
+    ASSERT_OK(status);
+
+    ASSERT_OK_AND_ASSIGN(trace_processor_redacted_,
+                         LoadTrace(GetRedactedTrace()));
+    ASSERT_OK_AND_ASSIGN(trace_processor_unredacted_,
+                         LoadTrace(GetSourceTrace()));
+  }
+
+  std::unique_ptr<trace_processor::TraceProcessor> trace_processor_unredacted_;
+  std::unique_ptr<trace_processor::TraceProcessor> trace_processor_redacted_;
+};
+
+TEST_F(PrunePerfEventsIntegrationTest, OnlyKeepsTargetProcessPerfSamples) {
+  // This query retrieves the total number of perf samples for target process
+  // in redacted trace
+  auto query =
```

Comment:
What is the query doing? Please make the name more meaningful and add a comment.